File: src/frontend/src/api/catalog.ts
import type { LocationArea, PublicCategory } from './types'

const publicCategoryLabels: Record<PublicCategory, string> = {
  ELECTRONICS: '电子产品',
  IDENTITY_CARD: '证件卡片',
  CLOTHING: '服饰配饰',
  STATIONERY: '学习用品',
  OTHER_CATEGORY: '其他',
}

const locationAreaLabels: Record<LocationArea, string> = {
  DORMITORY: '宿舍区',
  CANTEEN: '食堂',
  TEACHING_BUILDING: '教学楼',
  SCIENCE_BUILDING: '科教楼',
  LIBRARY: '图书馆',
}

function reverse<T extends string>(labels: Record<T, string>) {
  return Object.fromEntries(Object.entries(labels).map(([value, label]) => [label, value])) as Record<string, T>
}

const publicCategoriesByLabel = reverse(publicCategoryLabels)
const locationAreasByLabel = reverse(locationAreaLabels)

export const publicCategoryLabel = (category: PublicCategory) => publicCategoryLabels[category]
export const publicCategoryFromLabel = (label: string) => publicCategoriesByLabel[label]
export const locationAreaLabel = (area: LocationArea) => locationAreaLabels[area]
export const locationAreaFromLabel = (label: string) => locationAreasByLabel[label]

export const publicCategoryOptions = Object.entries(publicCategoryLabels).map(([value, label]) => ({ value: value as PublicCategory, label }))
export const locationAreaOptions = Object.entries(locationAreaLabels).map(([value, label]) => ({ value: value as LocationArea, label }))

File: src/backend/app/items/catalog.py
from app.db.enums import ItemType, LocationArea, PublicCategory


_LOCATION_LABELS = {
    LocationArea.DORMITORY: "宿舍区",
    LocationArea.CANTEEN: "食堂",
    LocationArea.TEACHING_BUILDING: "教学楼",
    LocationArea.SCIENCE_BUILDING: "科教楼",
    LocationArea.LIBRARY: "图书馆",
}


def item_type_for(category: PublicCategory) -> ItemType:
    if category is PublicCategory.IDENTITY_CARD:
        return ItemType.IDENTITY_DOCUMENT
    return ItemType.OTHER


def location_public_for(area: LocationArea) -> str:
    return _LOCATION_LABELS[area]


def build_public_embedding_text(
    *, name_public: str, description_public: str, location_public: str
) -> str:
    return "\n".join(
        (name_public.strip(), description_public.strip(), location_public.strip())
    )
